testsuite/MDAnalysisTests/coordinates/test_rdkit.py

```diff
@@ -20,28 +20,104 @@
 # MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
 # J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
 #
-import warnings
 
 import pytest
 import MDAnalysis as mda
+from MDAnalysis.topology.guessers import guess_atom_element
 import numpy as np
 from numpy.testing import (assert_equal,
                            assert_almost_equal)
 
-from MDAnalysisTests.datafiles import mol2_molecule
+from MDAnalysisTests.datafiles import mol2_molecule, PDB_full, GRO
+from MDAnalysisTests.util import block_import, import_not_available
 
-Chem = pytest.importorskip("rdkit.Chem")
-AllChem = pytest.importorskip("rdkit.Chem.AllChem")
 
-def mol2_mol():
-    return Chem.MolFromMol2File(mol2_molecule, removeHs=False)
+@block_import('rdkit')
+class TestRequiresRDKit(object):
+    def test_converter_requires_rdkit(self):
+        u = mda.Universe(mol2_molecule)
+        with pytest.raises(ImportError,
+                           match="RDKit is required for the RDKitConverter"):
+            u.atoms.convert_to("RDKIT")
 
-def smiles_mol():
-    mol = Chem.MolFromSmiles("CCO")
-    mol = Chem.AddHs(mol)
-    cids = AllChem.EmbedMultipleConfs(mol, numConfs=3)
-    return mol
 
+try:
+    from rdkit import Chem
+    from rdkit.Chem import AllChem
+    from MDAnalysis.coordinates.RDKit import (
+        RDATTRIBUTES,
+        _add_mda_attr_to_rdkit,
+        _infer_bo_and_charges,
+        _standardize_patterns,
+        _rebuild_conjugated_bonds,
+        _set_atom_property,
+        _reassign_props_after_reaction,
+    )
+except ImportError:
+    def mol2_mol():
```

Comment:
Just curious--the machinery that skips the tests when rdkit is absent is not sufficient to alleviate the need for "mocking" these functions?

Is this because of the trickery with using the functions in fixtures, so there's an issue where the skipping hasn't happened yet at test collection time?

That probably means there's a bit too much magic going on here, but I can see why I guess.

[Indirect parametrization](https://docs.pytest.org/en/latest/example/parametrize.html#indirect-parametrization) to allow fixture usage in the parametrization may be something to aim for eventually, even if this is "ok" here for now, to cut down on this top-level "mocking."

Reply:
Yup exactly, but I'm using indirect param now thanks to your comment 😃 I somehow missed this when looking at the pytest docs.
I think it's much better now